activerecord/lib/active_record/associations/belongs_to_association.rb

```diff
@@ -5,7 +5,15 @@ module Associations
     # = Active Record Belongs To Association
     class BelongsToAssociation < SingularAssociation #:nodoc:
       def handle_dependency
-        target.send(options[:dependent]) if load_target
+        return unless load_target
+
+        case options[:dependent]
+        when :destroy
+          target.destroy
+          raise ActiveRecord::Rollback unless target.destroyed?
```

Comment:
Does this get rescued further up the chain, or does it leak out of a `model.destroy`? We should only be raising an exception when the original caller is `destroy!`.

Reply:
Yes, it is rescued in rails/activerecord/lib/active_record/transactions.rb line 387:

```ruby
      383:       self.class.transaction do
      384:         add_to_transaction
      385:         begin
      386:           status = yield
      387:         rescue ActiveRecord::Rollback
=> 388:           clear_transaction_record_state
      389:           status = nil
      390:         end
      391:
      392:         raise ActiveRecord::Rollback unless status
```

Reply:
Yes, it gets rescued on rails/activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb line 256:

```ruby
   254:           transaction_manager.within_new_transaction(isolation: isolation, joinable: joinable) { yield }
   255:         end
=> 256:       rescue ActiveRecord::Rollback
   257:         # rollbacks are silently swallowed
   258:       end
```
handle_dependency is called within the context of an after_destroy callback (rails/activerecord/lib/active_record/associations/builder/belongs_to.rb line 133)